UI, Game and Math changes throughout development of It's Absolutely safe

Bromvlieg opened the pull request:
While using Mainframe as engine for It's Absolutely Safe we worked on the following features

- Ultralight/CEF integration with UI library, this required scancode to be added to correctly interpret characters from typing
- Matrix was not tested earlier, using it practice needed a few cleanup changes to finish porting it from TomatoLib
- Extending the Game namespace regarding players and a game server
- Fixing a bug where the OpenGL window context was not cleared after swapping buffers
- First part of removing shared pointers from entities. We've implemented a Reference object to have correct ownership in It's Absolutely safe

Hazzytje commented:
![image](https://user-images.githubusercontent.com/2007879/113170517-eb6bab00-9246-11eb-8881-89e46ad471ed.png)

Bromvlieg committed:
add .h filetype to cmake for IDE's like Visual Studio and VSCode

Bromvlieg committed:
UI: add scancode to key up and down

Bromvlieg committed:
UI: add `super` modifier key enum

Bromvlieg committed:
math: matrix cleanup

- add getValues()
- make constructor use const std::array&
- remove empty deconstructor

Bromvlieg committed:
game: clear screenbuffer on new frame

Bromvlieg committed:
game: add networking message base classes

Bromvlieg committed:
game:: first step of moving away from shared pointers to unique

Bromvlieg committed:
game: implement player functionality for servers

Bromvlieg commented:
![](https://media3.giphy.com/media/unQ3IJU2RG7DO/giphy.gif)
